native/src/components/PinchPanImage.tsx

```diff
@@ -1,338 +1,84 @@
+  const [imageDimensions, setImageDimensions] = useState<Dimensions>()
 
-    this.lastOffset = { x: 0, y: 0 }
-    this.onPanGestureEvent = Animated.event(
-      [
-        {
-          nativeEvent: {
-            translationX: this.translateX,
-            translationY: this.translateY,
-          },
-        },
-      ],
-      { useNativeDriver: USE_NATIVE_DRIVER },
-    )
-    this.translateXOffset = new Animated.Value(0)
-    this.translateYOffset = new Animated.Value(0)
-
-    // Make translation depend on scale and offset
-    this.scaledTranslateXWithOffset = Animated.divide(Animated.add(this.translateX, this.translateXOffset), this.scale)
-    this.scaledTranslateYWithOffset = Animated.divide(Animated.add(this.translateY, this.translateYOffset), this.scale)
-
-    // Setup: Initialize image dimensions
-    const { uri, onError } = props
+  useEffect(() => {
     Image.getSize(
       uri,
-      (width, height) => {
-        this.setState(state => ({ ...state, imageDimensions: { width, height } }))
-      },
-      (error: unknown) => {
-        onError(error)
-      },
+      (width, height) => setImageDimensions({ width, height }),
+      err => onError(err),
     )
-  }
+  }, [uri, onError])
 
-  private onImageLoadError = (error: NativeSyntheticEvent<ImageErrorEventData>) => {
-    const { onError } = this.props
-    onError(error.nativeEvent.error)
-  }
+  const prevScale = useSharedValue(1)
+  const scale = useSharedValue(1)
 
-  /**
-   * Moves the image along some axis back into the view. This animates the translateX and translateY properties
-   * if they are out of bounds
-   */
-  private fixBound(axis: 'x' | 'y', minValue: number, newValue: number) {
-    this.setState(state => ({ ...state, interactive: false }))
+  const translationX = useSharedValue(0)
+  const translationY = useSharedValue(0)
+  const prevTranslationX = useSharedValue(0)
+  const prevTranslationY = useSharedValue(0)
 
-    const animation: Animated.CompositeAnimation = Animated.timing(axis === 'x' ? this.translateX : this.translateY, {
-      toValue: minValue - newValue,
-      duration: ANIMATION_DURATION,
-      useNativeDriver: USE_NATIVE_DRIVER,
-    })
+  const animatedStyles = useAnimatedStyle(() => ({
+    transform: [{ translateX: translationX.value }, { translateY: translationY.value }, { scale: scale.value }],
+  }))
 
-    animation.start(({ finished }) => {
-      if (finished) {
-        if (axis === 'x') {
-          this.lastOffset.x = minValue
-          this.translateXOffset.setValue(minValue)
-          this.translateX.setValue(0)
-        } else {
-          this.lastOffset.y = minValue
-          this.translateYOffset.setValue(minValue)
-          this.translateY.setValue(0)
-        }
-      }
-      this.setState(state => ({ ...state, interactive: true }))
+  const pinchGesture = Gesture.Pinch()
+    .onStart(() => {
+      prevScale.value = scale.value
+    })
+    .onUpdate(e => {
+      scale.value = prevScale.value * e.scale
```

Comment:
🙃 Please avoid abbreviations here, it is not clear if `e` stands for `event` or `error` here. Also, you could get rid of the brackets.

Reply:
Good catch, thanks!